GoodNight/Goodnight.cs

```diff
@@ -211,8 +172,118 @@ private void OnNPCKilled(NpcKilledEventArgs args)
+                            $"禁止召唤怪物时段: " +
+                            $"[c/DF95EC:{Config.Time.Start}] — [c/FF9187:{Config.Time.Stop}]", Color.AntiqueWhite);
+                    }
+                }
+                else
+                {
+                    if (NpcList)
+                        args.Handled = false;
+                    Main.npc[args.NpcId].active = true;
+                }
+            }
         }
         #endregion
 
+        #region 开启禁止刷怪区方法
+        private void OnRegion(NpcSpawnEventArgs args)
+        {
+            int PlayerCount = TShock.Utils.GetActivePlayerCount();
+            bool NpcList = Config.Npcs.Contains(Main.npc[args.NpcId].netID);
+            bool NoPlr = PlayerCount <= Config.MaxPlayers && Config.MaxPlayers > 0;
+            var NpcListInfo = string.Join(", ", Config.Npcs.Select(x => TShock.Utils.GetNPCById(x)?.FullName));
+
+            if (DateTime.Now.TimeOfDay >= Config.Time.Start && DateTime.Now.TimeOfDay < Config.Time.Stop)
+            {
+                if (NoPlr)
+                {
+                    if (InRegion("召唤区"))
+                    { 
+                        if (NpcList)
+                        {
+                            args.Handled = false;
+                            Main.npc[args.NpcId].active = true;
+                            TShock.Utils.Broadcast($"【宵禁】当前[c/338AE1:服务器]存在 [c/FF3A4B:{PlayerCount}/{Config.MaxPlayers}]个玩家! \n检测到所有玩家已在【[c/FF3A4B:召唤区]】允许召唤以下怪物：\n" +
+                                $"[c/6EABE9:{NpcListInfo}]", Color.Aquamarine);
+                        }
+                    }
+                    else
+                    {
+                        if (NpcList)
+                        {
+                            args.Handled = true;
+                            Main.npc[args.NpcId].active = false;
+                            TShock.Utils.Broadcast($"【宵禁】不在召唤区无法生成怪物\n" +
+                                $"在线人数少于[c/FF3A4B:{Config.MaxPlayers}人]且不处于[c/338AE1:允许召唤区]", Color.AntiqueWhite);
+                        }
+                    }
+                }
+                else
+                {
+                        if (NpcList)
+                            args.Handled = false; 
+                    Main.npc[args.NpcId].active = true;
+                }
+            }
+        } 
+        #endregion
+
+        #region 判断所有玩家在Region方法
+        private static bool Region(float x, float y, string regionName)
+        {
+            var region = TShock.Regions.GetRegionByName(regionName);
+            if (region == null) 
+                return false;
+            return 
+                x >= region.Area.X && x <= region.Area.X + region.Area.Width &&
+                y >= region.Area.Y &&  y <= region.Area.Y + region.Area.Height;
+        }
+
+        private static bool InRegion(string regionName)
+        {
+            for (int i = 0; i < Main.maxPlayers; i++)
```

Comment:
foreach(var plr in TShock.Players)
{
     if(plr != null && plr.Active && plr.CurrentRegion != null && plr.CurrentRegion.Name == regionName)
}